test/extensions/filters/http/grpc_json_reverse_transcoder/filter_test.cc

```diff
@@ -0,0 +1,741 @@
+TEST_F(GrpcJsonReverseTranscoderFilterTest, RequestMissingPathParam) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "POST"},
+                                         {":path", "/bookstore.Bookstore/CreateShelfBodyWildcard"},
+                                         {"content-type", "application/grpc"}};
+  EXPECT_EQ(Http::FilterHeadersStatus::StopIteration, filter_.decodeHeaders(headers, false));
+
+  EXPECT_CALL(decoder_callbacks_, sendLocalReply(Http::Code::BadRequest, _, _, _, _));
+  bookstore::CreateShelfRequest request;
+  request.mutable_shelf()->set_theme("Kids");
+  auto request_data = Grpc::Common::serializeToGrpcFrame(request);
+  EXPECT_EQ(Http::FilterDataStatus::StopIterationNoBuffer, filter_.decodeData(*request_data, true));
+}
+
+// Test request transcoding with gRPC message missing the payload for the HTTP request.
+TEST_F(GrpcJsonReverseTranscoderFilterTest, TranscodeMissingBody) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "POST"},
+                                         {":path", "/bookstore.Bookstore/UpdateBook"},
+                                         {"content-type", "application/grpc"}};
+  EXPECT_EQ(Http::FilterHeadersStatus::StopIteration, filter_.decodeHeaders(headers, false));
+
+  EXPECT_CALL(decoder_callbacks_, sendLocalReply(Http::Code::BadRequest, _, _, _, _));
+
+  bookstore::UpdateBookRequest request;
+  request.set_shelf(12345);
+  auto request_data = Grpc::Common::serializeToGrpcFrame(request);
+  EXPECT_EQ(Http::FilterDataStatus::StopIterationNoBuffer, filter_.decodeData(*request_data, true));
+}
+
+// Test request transcoding with data decoding divided into multiple frames.
+TEST_F(GrpcJsonReverseTranscoderFilterTest, RequestWithMultipleDataFrames) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "POST"},
+                                         {":path", "/bookstore.Bookstore/UpdateBook"},
+                                         {"content-type", "application/grpc"}};
+  EXPECT_EQ(Http::FilterHeadersStatus::StopIteration, filter_.decodeHeaders(headers, false));
+
+  bookstore::UpdateBookRequest request;
+  request.set_shelf(12345);
+  request.mutable_book()->set_id(123);
+  request.mutable_book()->set_title("Kids book");
+  request.mutable_book()->set_author("John Doe");
+  std::string expected_request = "{\"author\":\"John Doe\",\"id\":\"123\",\"title\":\"Kids book\"}";
+  auto request_data = Grpc::Common::serializeToGrpcFrame(request);
+  EXPECT_EQ(Http::FilterDataStatus::StopIterationNoBuffer,
+            filter_.decodeData(*request_data, false));
+  request_data->add("");
+  EXPECT_EQ(Http::FilterDataStatus::Continue, filter_.decodeData(*request_data, true));
+  EXPECT_EQ(headers.getPathValue(), "/shelves/12345/books/123");
+  EXPECT_EQ(request_data->toString(), expected_request);
+  EXPECT_EQ(headers.getContentLengthValue(), std::to_string(expected_request.size()));
+}
+
+// Test request transcoding where request type is google.api.HttpBody
```

Comment:
```suggestion
// Test request transcoding where request type is `google.api.HttpBody`
```

not certain, but i think marking as literal will fix _this_ spellcheck failure

Reply:
Fixed